cmake/Codegen.cmake

```diff
@@ -53,35 +53,35 @@ install(FILES ${CMAKE_BINARY_DIR}/caffe2/core/macros.h
 
 # ---[ ATen specific
 if (NOT BUILD_ATEN_MOBILE)
-  SET(OPT_FLAG "-O3 ")
-  IF(MSVC)
-    SET(OPT_FLAG "/Ox /fp:strict ")
-  ENDIF()
-  SET(VCOMP_LIB "vcomp")
+  if(MSVC)
+    set(OPT_FLAG "/OX /fp:strict ")
+  else()
+    set(OPT_FLAG "-O3 ")
+  endif()
+  SET(VCOMP_LIB vcomp$<$<CONFIG:Debug>:d>)
```

Comment:
This one is not expanded as I can see this line in the log.
```
07:10:41 cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_dll --intdir=caffe2\CMakeFiles\caffe2.dir --manifests  -- C:\PROGRA~2\MIB055~1\2017\COMMUN~1\VC\Tools\MSVC\1411~1.255\bin\Hostx64\x64\link.exe /nologo @CMakeFiles\caffe2.rsp  /out:bin\caffe2.dll /implib:LIB\caffe2.lib /pdb:bin\caffe2.pdb /dll /version:0.0 /INCREMENTAL:NO /NODEFAULTLIB:vcomp$<$<CONFIG:Debug>:d>   && cd ."
07:10:41 && was unexpected at this time.
```

Reply:
Thanks.
I wanted to avoid all the nasty if/else checking everywhere but seems like I'll have to do it.

Is it possible at all to bump the CMake minimum version or is that already new enough (3.5)?

Reply:
I'm afraid this is not the issue here. We are using CMake 3.12.